### PR TITLE
Feature: request validators

### PR DESCRIPTION
**Pull request by Samuel-B-D**
Added an optional feature flag to automatically validate request Json/Form payloads with the https://github.com/Keats/validator crate.

The macro is not re-exported through saphir because of path resolution issue in the resulting generated code (I had a long discussion with some macro experts on the official rust discord and ended up to the conclusion that it isn't feasible without adding some significant changes inside the validator crate itself, which wouldn't be welcome by them).

So, the usage is to import the crate in the project which is using saphir, implement validators as per the validator crate documentation, and add the saphir feature flag `validate-requests` to cause saphir's macro generation to generate validation code using the Validator trait. This flag is an optional flag not included in the `full` flag.

The feature flag by default enable validation on all Json/Form request parameters. This will trigger compile error if any request payload is added without proper validation. A new macro attribute was added, `#[validator(...)`, which for now allow to exclude validation on specific parameter by using the parameter name.
For example:

```rust
#[post("/")]
#[validator(excude(req))]
async fn my_route(&self, req: Json<RoutePayload>) -> Result<(), MyError> {
   //...
}
```
The exclude attribute take a comma-separated list of parameter names to be excluded.

@LucFauvel please review this.

### CODE REVIEW

**Comment by Samuel-B-D** [comment]
Link to the beginning of the discussion I had on the official Rust Discord: https://discord.com/channels/273534239310479360/512792629516173323/996427692389978193

**Review by LucFauvel — APPROVED** [review]

File: saphir_macro/Cargo.toml [hunk 1]
```diff
@@ -16,8 +16,13 @@ license = "MIT"
...
+full = ["validate-requests"]
+validate-requests = []
+
 [dependencies]
 proc-macro2 = "1.0"
 quote = "1.0"
-syn = { version = "1.0.16", features = ["full", "extra-traits"] }
-http = "0.2"
+syn = { version = "1.0.16", features = ["full"] }
+http = "0.2"
```
Comment (richerarc) [inline saphir_macro/Cargo.toml:28]: manque un newline

**Review by richerarc — APPROVED** [review]

File: saphir_macro/Cargo.toml [hunk 2]
```diff
@@ -16,8 +16,13 @@ license = "MIT"
 name = "saphir_macro"
 proc-macro = true
 
+[features]
+default = []
+full = ["validate-requests"]
```
[the remark this answers is not in the document]
Comment (Samuel-B-D) [reply saphir_macro/Cargo.toml:21]: `default` is the default one that will be used by cargo (`validate-requests` *must* not be enabled by default)
`full` is just a standard feature usually found on crate, which is alrealy found on our main crate.

[the remark this answers is not in the document]
Comment (Samuel-B-D) [reply saphir_macro/Cargo.toml:21]: ( `validate-requests` is the feature that's checked for in the added macro code )

File: saphir_macro/src/controller/handler.rs [hunk 3]
```diff
@@ -338,309 +391,368 @@ impl HandlerRepr {
...
+                                _ => return Err(Error::new_spanned(openapi_attributes, "Invalid openapi attribute")),
                             }
-                        } else {
-                            let method =
-                                Method::from_str(ident.to_string().to_uppercase().as_str()).map_err(|_e| Error::new_spanned(ident, "Invalid HTTP method"))?;
+                        }
+                    } else if ident == "validator" {
+                        #[cfg(not(feature = "validate-requests"))]
+                        {
+                            return Err(Error::new_spanned(ident, "This require the \"validate-requests\" feature flag in Saphir"));
```
[the remark this answers is not in the document]
Comment (Samuel-B-D) [reply saphir_macro/src/controller/handler.rs]: fixed

File: saphir_macro/Cargo.toml (hunk 1 above)
Comment (Samuel-B-D) [reply saphir_macro/Cargo.toml]: fixed

**Pull request merged**